api/v1/controllers/samples.js

```diff
@@ -94,10 +94,9 @@ module.exports = {
    */
   findSamples(req, res, next) {
     // Check if Cache is on for Wildcard Sample query
-    if (featureToggles.isFeatureEnabled('cacheGetSamplesWildcard')) {
-      const query = req.query.name;
-      helper.cacheEnabled =
-        query ? (query.indexOf('*') > -1) ? true : false : false;
+    if (featureToggles.isFeatureEnabled('cacheGetSamplesByNameWildcard')) {
+      const query = req.query.name || '';
```

Comment:
I think I was not clear in my comments. You do not need to do `|| ''`., just the change you have on line 99 is enough.